### PR TITLE
Do 2nd spring-boot-practise

### PR DESCRIPTION
**Pull request by khorkovenko**

### CODE REVIEW

File: store/src/main/java/com/book/store/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            return ResponseEntity.ok(shoppingCart);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @Operation(summary = "Add book to the shopping cart")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Book added successfully"),
+            @ApiResponse(responseCode = "400", description = "Invalid request parameters")
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/controller/ShoppingCartController.java]: For what you need this?

File: store/src/main/java/com/book/store/controller/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            return ResponseEntity.ok(shoppingCart);
+        } catch (InvalidRequestException e) {
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    @Operation(summary = "Update quantity of a book in the shopping cart")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Quantity updated successfully"),
+            @ApiResponse(responseCode = "400", description = "Invalid request parameters")
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/controller/ShoppingCartController.java]: Is it make any sense?

File: store/src/main/java/com/book/store/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import com.book.store.repo.ShoppingCartRepository;
+import com.book.store.service.ShoppingCartService;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
```

File: store/src/main/java/com/book/store/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    @Operation(summary = "Get user's shopping cart by ID")
+    @GetMapping
+    public ResponseEntity<ShoppingCart> getUserShoppingCart(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartService.getUserShoppingCart(userId);
+
+        if (shoppingCart != null) {
+            return ResponseEntity.ok(shoppingCart);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/controller/ShoppingCartController.java]: Why you make validation here and where your dto?

File: store/src/main/java/com/book/store/controller/ShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            Long userId
+    ) {
+        try {
+            ShoppingCart shoppingCart = shoppingCartService
+                    .addBookToCart(userId, request.getBookId(), request.getQuantity());
+
+            return ResponseEntity.ok(shoppingCart);
+        } catch (InvalidRequestException e) {
+            return ResponseEntity.badRequest().build();
+        }
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/controller/ShoppingCartController.java]: You wrap every method by try catch, is it important use it here?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Review by koval-kn — APPROVED** [review]

**Pull request closed without merging**